ddtrace/contrib/botocore/services/kinesis.py

```diff
@@ -134,48 +136,86 @@ def patched_kinesis_api_call(original_func, instance, args, kwargs, function_var
-    ) as span:
-        set_patched_api_call_span_tags(span, instance, args, params, endpoint_name, operation)
-
-        if args:
-            if config.botocore["distributed_tracing"]:
-                try:
-                    if endpoint_name == "kinesis" and operation in {"PutRecord", "PutRecords"}:
-                        inject_trace_to_kinesis_stream(
-                            params, span, pin=pin, data_streams_enabled=config._data_streams_enabled
-                        )
-                        span.name = schematize_cloud_messaging_operation(
-                            trace_operation,
-                            cloud_provider="aws",
-                            cloud_service="kinesis",
-                            direction=SpanDirection.OUTBOUND,
-                        )
-                except Exception:
-                    log.warning("Unable to inject trace context", exc_info=True)
+    message_received = False
+    func_run = False
+    func_run_err = None
+    child_of = None
+    start_ns = None
+    result = None
 
+    if operation == "GetRecords":
         try:
+            start_ns = time_ns()
+            func_run = True
             result = original_func(*args, **kwargs)
+        except Exception as e:
+            func_run_err = e
+        if result is not None and "Records" in result and len(result["Records"]) >= 1:
+            message_received = True
+            if config.botocore.propagation_enabled:
+                child_of = extract_DD_context(result["Records"])
 
-            if endpoint_name == "kinesis" and operation == "GetRecords" and config._data_streams_enabled:
-                try:
-                    record_data_streams_path_for_kinesis_stream(pin, params, result, span)
-                except Exception:
-                    log.debug("Failed to report data streams monitoring info for kinesis", exc_info=True)
-
-            set_response_metadata_tags(span, result)
-            return result
-
-        except botocore.exceptions.ClientError as e:
-            # `ClientError.response` contains the result, so we can still grab response metadata
-            set_response_metadata_tags(span, e.response)
-
-            # If we have a status code, and the status code is not an error,
-            #   then ignore the exception being raised
-            status_code = span.get_tag(http.STATUS_CODE)
-            if status_code and not config.botocore.operations[span.resource].is_error_code(int(status_code)):
-                span._ignore_exception(botocore.exceptions.ClientError)
-            raise
+    """
+    We only want to create a span for the following cases:
+        - not func_run: The function is not `getRecords` and we need to run it
+        - func_run and message_received: Received a message when polling
+        - config.empty_poll_enabled: We want to trace empty poll operations
+    """
+    if (func_run and message_received) or config.empty_poll_enabled or not func_run:
```

Comment:
Should we do this?

```suggestion
    if (func_run and message_received) or (config.empty_poll_enabled and not func_run):
```

Reply:
No, this doesn't work since:

- If we have `config.empty_poll_enabled=false`, this will never run, regardless of `func_run`